core/src/main/java/org/apache/iceberg/rest/requests/RegisterTableRequestParser.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.catalog.TableIdentifierParser;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+public class RegisterTableRequestParser {
+
+  private static final String IDENTIFIER = "identifier";
+  private static final String METADATA_FILE_LOCATION = "metadata-file-location";
+
+  private RegisterTableRequestParser() {}
+
+  public static String toJson(RegisterTableRequest request) {
+    return toJson(request, false);
+  }
+
+  public static String toJson(RegisterTableRequest request, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(request, gen), pretty);
+  }
+
+  public static void toJson(RegisterTableRequest request, JsonGenerator gen) throws IOException {
+    Preconditions.checkArgument(null != request, "Invalid register table request: null");
+
+    gen.writeStartObject();
+
+    gen.writeFieldName(IDENTIFIER);
+    TableIdentifierParser.toJson(request.identifier(), gen);
```

Comment:
As @nastra suggested (https://github.com/apache/iceberg/pull/6512#discussion_r1230706545 and  https://github.com/apache/iceberg/pull/6512#discussion_r1230708833). I modified it to send identifier.
> I think it's better to track the TableIdentifier instead of a plain string

I see that `UpdateTableRequestParser` is doing the same thing. 

https://github.com/apache/iceberg/blob/9ffb762153ce050c2d961327886ef9a41b2d6145/core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequestParser.java#L56-L57

Reply:
Hi @rdblue | @nastra, I have changed the `RegisterTableRequest` to have table-name instead identifier.